plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -741,11 +741,35 @@ public static Set<HivePrivilegeInfo> parsePrivilege(PrivilegeGrantInfo userGrant
     }
 
     public static HiveBasicStatistics getHiveBasicStatistics(Map<String, String> parameters)
+    {
+        return getHiveBasicStatistics(parameters, false);
+    }
+
+    public static HiveBasicStatistics getHiveBasicStatistics(Map<String, String> parameters, boolean useSparkFallback)
     {
         OptionalLong numFiles = toLong(parameters.get(NUM_FILES));
         OptionalLong numRows = toLong(parameters.get(NUM_ROWS));
         OptionalLong inMemoryDataSizeInBytes = toLong(parameters.get(RAW_DATA_SIZE));
         OptionalLong onDiskDataSizeInBytes = toLong(parameters.get(TOTAL_SIZE));
+
+        if (useSparkFallback) {
+            if ((numFiles.isEmpty() || numFiles.getAsLong() == 0)) {
+                OptionalLong sparkNumFiles = toLong(parameters.get(ThriftSparkMetastoreUtil.SPARK_SQL_STATS_PREFIX + NUM_FILES));
+                numFiles = sparkNumFiles.isPresent() ? sparkNumFiles : numFiles;
```

Comment:
Instead of flipping back and forth between hive and spark stats for each parameter, we should use spark stats for all 4 parameters only when hive row count is absent or 0 and spark row count is present and non-zero.